docs: [FC-0074] index, references and concepts style review (#262)
openedx/openedx-filters: docs/concepts/openedx-filters.rst

   :align: center

Components
============

* **Application (caller):** The component that calls the filter during its execution, triggering the pipeline to process the input data. Developers may have added this call to a part of the application to include different behaviors. E.g., a user enrolls in a course, triggering the `CourseEnrollmentStarted filter`_.
* **OpenEdxPublicFilter:** This class implements all methods used to manage the execution of the filter.
* **PipelineStep1...N:** The pipeline steps that are executed in sequence, each processing the input data and returning potentially modified data. The developer defines these steps to introduce additional behaviors.  E.g., a pipeline step that checks user eligibility for enrollment.

Workflow
===========

#. An application component (caller) invokes the filter during its execution by calling the ``run_filter`` method implemented by its :term:`filter definition<Filter Definition>`.

> #. The caller passes the input data to the filter through the ``run_filter`` method. This data is an in-memory platform object that the filter will process.

Reviewer: ```suggestion
#. The caller passes the input data to the filter through the ``run_filter`` method. These data are in-memory platform objects that the filter will process.
```
